paasta_tools/setup_istio_mesh.py

```diff
@@ -185,11 +187,52 @@ def setup_paasta_namespace_services(
             log.warning(f"{err} while setting up k8s service for {namespace}")
             status = False
 
-    return status
+    return (status, api_updates)
+
+
+def cleanup_kube_svc(
+    kube_client: KubeClient,
+    smartstack_namespaces: AbstractSet,
+    existing_svc_names: Set[str],
+    api_updates: int = 0,
+) -> Tuple[bool, int]:
+    sanitized_smarstack_namespaces = {
+        sanitise_kubernetes_service_name(namespace)
+        for namespace in smartstack_namespaces
+    }
+
+    status = True
+    if not existing_svc_names:
+        return status, api_updates
+
+    for svc in existing_svc_names.difference(sanitized_smarstack_namespaces):
+        if svc == UNIFIED_K8S_SVC_NAME:
+            continue
+
+        if RATE_LIMIT > 0 and api_updates >= RATE_LIMIT:
+            log.info(
+                f"Not cleaning up further services as we reached the limit ({api_updates})"
+            )
+            break
+
+        try:
+            log.info(
+                f"Garbage collecting {svc} since there is no reference in services.yaml"
+            )
+            kube_client.core.delete_namespaced_service(
+                name=svc, namespace=PAASTA_NAMESPACE
+            )
+            api_updates += 1
+
+        except Exception as err:
+            log.warning(f"{err} while trying to garbage collect {svc}")
+            traceback.print_exc()
```

Comment:
super-nit: this could also be written as 
```suggestion
        except Exception:
            log.exception(f"Error encountered while trying to garbage collect {svc}")
```
so that we don't need to manually print out the exception

Reply:
Thanks, this is quite nit